Recommend Square for all (online & offline)

elazzabi opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #55211

Make sure that Square is shown for all merchant (it was before shown for in-person merchants only).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `Tools > WCA Test Helper` and then the `Tools` tab
2. Execute the `Reset Onboarding Wizard`

<img width="1465" alt="Screenshot 2025-02-10 at 16 40 21" src="https://github.com/user-attachments/assets/1e29f86f-4626-4e14-97eb-886de4d348ca" />

3. Click on WooCommerce in the left sidebar
4. You should get the onboarding wizard, do not skip it. Click on `Set up my store`

<img width="838" alt="Screenshot 2025-02-10 at 16 41 27" src="https://github.com/user-attachments/assets/e448c1c6-a66e-40b6-9142-8c96ca4e4422" />

5. When asked if you sell online, select `Yes, I'm selling online`

<img width="702" alt="Screenshot 2025-02-10 at 16 41 35" src="https://github.com/user-attachments/assets/7c549231-7f75-46e1-9048-0c3851160db5" />

6. Continue the onboarding process
7. Once finished, go to `WooCommerce > Settings > Payments`
8. If your store is in a country not supported by Square, you can switch the country from the top-right country dropdown. Select `Canada`, `US` or any other supported country.
9. Make sure Square is in the list and after Stripe

(This is Canada)
<img width="1427" alt="Screenshot 2025-02-10 at 16 44 09" src="https://github.com/user-attachments/assets/30a94f69-775f-4e2d-a86b-de9c92243d50" />

(This is the US)
<img width="1441" alt="Screenshot 2025-02-10 at 16 43 57" src="https://github.com/user-attachments/assets/924aa80d-3d7d-4571-9767-ca63726cab53" />

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Make sure Square is shown for all merchants (online & in-person) where supported

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

elazzabi committed:
Remove extra check for in person

elazzabi committed:
Remove the is_extension_allowed function completely

elazzabi committed:
Merge branch 'trunk' into fix/square-for-all

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

vladolaru commented on plugins/woocommerce/src/Internal/Admin/Suggestions/PaymentExtensionSuggestions.php:
```diff
@@ -1722,11 +1722,6 @@ public function get_country_extensions( string $country_code, string $context =
 				continue;
 			}
 
-			// Determine if the extension should be included based on the store's state, the provided country and context.
```
I would keep this as a means to provide a ready-made place for per-extension logic. I know the `is_extension_allowed` method will end up with just returning `true`, but I think we gain more than we lose. WDYT?

elazzabi replied:
Yeah, I 100% agree. The first solution I did was returning `true` like you said but then I removed the whole function in 06416e0bf2e0b0f163170db90b0990955e454e8c. However, I agree with you, let's leave it there for now

vladolaru reviewed (commented):
Thank you for working on this, @elazzabi!

I think this PR is a good opportunity to rename the `SQUARE_IN_PERSON` constant and its value to just Square since we are not into legacy territory yet.

elazzabi committed:
Revert "Remove the is_extension_allowed function completely"

This reverts commit 06416e0bf2e0b0f163170db90b0990955e454e8c.

elazzabi committed:
Use SQUARE instead of SQUARE_IN_PERSON

elazzabi committed:
Merge branch 'fix/square-for-all' of https://github.com/woocommerce/woocommerce into fix/square-for-all

elazzabi committed:
Change Square in PaymentProviders

elazzabi committed:
Fix tests

vladolaru commented on plugins/woocommerce/src/Internal/Admin/Suggestions/PaymentExtensionSuggestions.php:
```diff
@@ -1844,17 +1844,6 @@ public function dismiss_incentive( string $incentive_id, string $suggestion_id,
 	 *              Defaults to true if there is no specific logic for the extension.
 	 */
 	private function is_extension_allowed( string $extension_id, string $country_code, string $context = '' ): bool { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.FoundAfterLastUsed
-
```
Maybe add a comment here about what to do. Something like:

```
// Add per-extension exclusion logic here.
```

elazzabi committed:
Fix linting issues

elazzabi committed:
Merge branch 'trunk' into fix/square-for-all

vladolaru reviewed (approved):
Things are looking great now! Left a minor comment, but pre-approving it!

elazzabi committed:
Add clarifying comments to PaymentExtensionSuggestions::is_extension_allowed

Add comments to clarify that the method is a placeholder for future per-extension exclusion logic and currently returns true to avoid excluding any extensions.

elazzabi committed:
Merge branch 'fix/square-for-all' of https://github.com/woocommerce/woocommerce into fix/square-for-all

elazzabi committed:
Merge branch 'trunk' into fix/square-for-all
